main/src/library/MutSet.flix

```diff
@@ -187,14 +193,14 @@ namespace MutSet {
     ///
     /// The function `f` must be pure.
     ///
-    pub def find(f: a -> Bool, s: MutSet[a]): Option[a] & Impure = findLeft(f, s)
+    pub def find(f: a -> Bool, s: mut MutSet[a, r]): Option[a] & reg(r) = findLeft(f, s)
```

Comment:
The `mut` keyword actually does not do anything, and I think rather than adding it, we should be removing it.

Later we will figure out what it does and then we can re-add it.